bin/stalePullRequest.js

```diff
@@ -1,92 +1,131 @@
+        })
+        .catch(function (err) {
+            console.error(err);
+        });
+}
+
+/**
+ * Bumps stale pull requests for all configured repositories.
+ *
+ * @param {Object} repositories The Settings.repositories object.
+ * @returns {Promise} A promise that resolves when the process is complete.
+ */
+function stalePullRequest(repositories) {
+    dateLog('Initiating `stalePullRequest` job.');
+    return Promise.each(Object.keys(repositories), function (repositoryName) {
+        var repositorySettings = repositories[repositoryName];
+        return stalePullRequest._processRepository(repositoryName, repositorySettings)
+            .catch(function (error) {
+                //Eat the error here so that all repositories are processed.
+                console.error(error);
+            });
+    });
+}
+
+/**
+ * Get all pull requests for the given repository by going sequentially requesting all pages
+ * from the GitHub API.
+ *
+ * @param {String} repositoryName Base url to list pull requests https://developer.github.com/v3/pulls/#list-pull-requests
+ * @param {RepositorySettings} repositorySettings The repository settings
+ * @return {Promise<Array<http.IncomingMessage | undefined>>} Promise to an array of incoming messages
+ */
+stalePullRequest._processRepository = function (repositoryName, repositorySettings) {
+    dateLog('Checking ' + repositoryName);
+    var pullRequests = [];
+
+    function processPage(response) {
+        var linkData = parseLink(response.headers.link);
+        pullRequests = pullRequests.concat(response.body);
+        // If we're at the last page
+        if (!Cesium.defined(linkData) || !Cesium.defined(linkData.next)) {
+            return Promise.each(pullRequests, function (pullRequest) {
+                return stalePullRequest._processPullRequest(pullRequest, repositorySettings);
+            });
+        }
+        // Otherwise, request the next page
+        return requestPromise.get({
+            url: linkData.next.url,
+            headers: repositorySettings.headers,
+            json: true,
+            resolveWithFullResponse: true
+        }).then(processPage);
+    }
+
+    return requestPromise.get({
+        url: 'https://api.github.com/repos/' + repositoryName + '/pulls?state=open&base=master',
+        headers: repositorySettings.headers,
+        json: true,
+        resolveWithFullResponse: true
+    }).then(processPage);
+};
+
+stalePullRequest._processPullRequest = function (pullRequest, repositorySettings) {
+    var commentsUrl = pullRequest.comments_url;
+    return requestPromise.get({
+        url: commentsUrl,
+        headers: repositorySettings.headers,
+        json: true,
+        resolveWithFullResponse: true
+    })
+        .then(function (response) {
+            var linkData = parseLink(response.headers.link);
+            if (Cesium.defined(linkData)) {
+                commentsUrl = linkData.last.url;
+            }

```

Comment:
It looks like we're always making another comment request even if there isn't another page.